## Problem 4.5

### 4.5.a
\begin{align}
\frac{dE_{rot}}{dt} &\propto \omega^6\\
E_{rot} =& \frac{1}{2} I \omega^2\\
\implies \frac{1}{2} I \frac{d}{dt}\omega^2 &= I \omega \dot{\omega} \propto \omega^6\\
\implies \frac{d\omega}{dt} &= C\omega^5\\
\implies \omega^{-5}\quad d\omega &= Cdt \\
\implies (\omega_0^{-4} - \omega^{-4}) = 4C(t-t_0)\\
\therefore \omega(t) = [\omega_0^{-4} - 4C(t-t_0)]^{-1/4}
\end{align}
where C is the constant of proportionality and $\omega_0$ is the initial frequency of rotation.

### 4.5.b
The age can be determined by
\begin{align}
t-t_0 = \frac{1}{4C}(\omega_0^{-4} - \omega^{-4})
\end{align}  
If we assume $\omega_0 \gg \omega$, we get
\begin{align}
t_0 - t = \frac{\omega^{-4}}{4C} = \frac{C\omega}{4C\dot{\omega}} = \frac{\omega}{4\dot{\omega}}
\end{align}
If we were to know the current value of $\omega$, we would be able to detremine the time scale of gravitational radiation and able to compare it with the time $1054yr-t_{now}$. If $t_0-t$ is less than that, gravitational radiation couldn't have contributed enough to the loss of energy in the supernova remnant.

## Problem 5.2

### 5.2.a
\begin{align}
\frac{dN}{dm} = am^{-2.35}
\end{align}
Total mass of the cluster is
\begin{align}
\int_{0.1}^{20} m\frac{dN}{dm}dm = M_{tot}\\
\implies \int_{0.1}^{20} am^{-1.35} dm = 10^6\\
\implies a = \frac{10^6\times 0.35}{0.1^{-0.35} - 20^{-0.35}}
\end{align}

In [1]:
val = 0.35*1e6/(0.1**(-.35) - 20**(-0.35))
print(val)

185355.80124054707


\begin{align}
\therefore a = 1.85 \times 10^5
\end{align}

### 5.2.b
Luminosity of sun of mass $M_\odot$ is $L_\odot$. Luminosity of any other star would thus be
\begin{align}
L = L_\odot\frac{M^4}{M_\odot^4} = L_\odot m^4
\end{align}
Total luminosity of the cluster is thus,
\begin{align}
L_{tot} = \int_{0.1}^{20} am^{4-2.35} dm L_\odot\\
\implies L_{tot} = a \frac{20^{2.65}-0.1^{2.65}}{2.65} L_\odot
\end{align}

In [2]:
v = 1.85e5*(20**2.65 - 0.1**2.65)/2.65
print(v)

195728918.40086555


\begin{align}
\therefore L_{tot} = 1.9 \times 10^8
\end{align}

\begin{align}
L_{>5} = a \frac{20^{2.65} - 5^{2.65}}{2.65} L_\odot
\end{align}
Thus, the fraction contributing to total luminosity is 
\begin{align}
N = \frac{L_{>5}}{L_{tot}} = \frac{20^{2.65}-5^{2.65}}{20^{2.65}-0.1^{2.65}}
\end{align}

In [3]:
v = (20**2.65 - 5**2.65)/(20**2.65 - 0.1**2.65)
print(v)

0.9746178908383528


\begin{align}
\therefore N = 0.975
\end{align}

### 5.2.c
The mean mass is given by total mass divided by total no. of stars,
\begin{align}
\bar{M} = \dfrac{\int_{0.1}^{20} am^{-1.35}dm}{\int_{0.1}^{20} am^{-2.35}dm} M_\odot\\
\implies \bar{M} = \frac{1.35}{0.35}\dfrac{20^{-0.35}-0.1^{-0.35}}{20^{-1.35}-0.1^{-1.35}} M_\odot
\end{align}

In [4]:
v = 1.35*(20**(-0.35)-0.1**(-0.35))/0.35/(20**(-1.35)-0.1**(-1.35))
print(v)

0.3255874466772297


\begin{align}
\therefore \bar{M} = 0.325 M_\odot
\end{align}

### 5.2.d
The lifetime of a solar mass star $M_\odot$ is 10Gyr. For any star this is 
\begin{align}
t = 10\textrm{Gyr} \quad m^{-2}
\end{align}

After 1Gyr, we should have
\begin{align}
0.1 = m^{-2}\\
\therefore m = \sqrt{10}
\end{align}

In [5]:
import numpy as np
v = np.sqrt(10)
print(v)

3.1622776601683795


The maximum mass star left after 1Gyr is thus
\begin{align}
M = 3.162 M_\odot
\end{align}

Total luminosity under the circumstance is thus
\begin{align}
L = a \dfrac{3.2^{2.65}-0.1^{2.65}}{2.65} L_\odot
\end{align}

In [6]:
v = 1.85e5*(3.2**2.65 - 0.1**2.65)/2.65
print(v)

1522399.788883329


\begin{align}
\therefore L = 1.52 \times 10^6 L_\odot
\end{align}